.github/workflows/build-and-test.yml

```diff
@@ -139,3 +139,167 @@ jobs:
+
+      - name: Build MSI
+        run: |
+          mkdir -p dist
+          gitRef=(${GITHUB_REF//// })
+          if [[ "${gitRef[1]}" = "tags" ]] ;
+          then
+            echo "TAG_NAME=${gitRef[2]}" >> $GITHUB_ENV
+          else
+            echo "TAG_NAME=" >> $GITHUB_ENV
+          fi
+          export VERSION_TAG="${TAG_NAME#v}"
+          make msi SKIP_COMPILE=true VERSION="${VERSION_TAG:-0.0.1.${{ github.run_number }}}"
+      
+      - name: Uploading msi build artifacts
+        uses: actions/upload-artifact@v2
+        with:
+          name: msi-build
+          path: ./dist/*.msi
+
+  windows-msi-validation:
+    name: windows-msi-validation
+    runs-on: windows-latest
+    needs: [windows-msi]
+    strategy:
+      matrix:
+        MODE: [ "agent", "gateway" ]
+    steps:
+      - name: Check out the codebase.
+        uses: actions/checkout@v2
+
+      - name: Downloading msi build
+        uses: actions/download-artifact@v2
+        with:
+          name: msi-build
+          path: ./dist
+
+      - name: Installation test
+        run: |
+          $ErrorActionPreference = 'Stop'
+          Set-PSDebug -Trace 1
+          $msi_path = Resolve-Path .\dist\splunk-otel-collector*.msi
+          $env:VERIFY_ACCESS_TOKEN = "false"
+          .\internal\buildscripts\packaging\installer\install.ps1 -access_token "testing123" -msi_path "$msi_path" -mode "${{ matrix.MODE }}"
+          Start-Sleep -s 30
+          Set-Location -Path "$env:ProgramFiles\Splunk\OpenTelemetry Collector"
+          Test-Path -Path ".\splunk-support-bundle.ps1"
+          Start-Process ".\otelcol.exe" -WindowStyle Hidden
+          powershell.exe -File "$env:ProgramFiles\Splunk\OpenTelemetry Collector\splunk-support-bundle.ps1" -t \tmp\splunk-support-bundle
+          Test-Path -Path ".\splunk-support-bundle.zip"
+          Test-Path -Path "\tmp\splunk-support-bundle\logs\splunk-otel-collector.log"
+          Test-Path -Path "\tmp\splunk-support-bundle\logs\splunk-otel-collector.txt"
+          Test-Path -Path "\tmp\splunk-support-bundle\logs\td-agent.log"
+          Test-Path -Path "\tmp\splunk-support-bundle\logs\td-agent.txt"
+          Test-Path -Path "\tmp\splunk-support-bundle\metrics\collector-metrics.txt"
+          Test-Path -Path "\tmp\splunk-support-bundle\metrics\df.txt"
+          Test-Path -Path "\tmp\splunk-support-bundle\metrics\free.txt"
+          Test-Path -Path "\tmp\splunk-support-bundle\metrics\top.txt"
+          Test-Path -Path "\tmp\splunk-support-bundle\zpages\tracez.html"
+          if ( "${{ matrix.MODE }}" -eq "agent" ) {
+            Test-Path -Path "\tmp\splunk-support-bundle\config\td-agent\td-agent.conf"
+            Test-Path -Path "\tmp\splunk-support-bundle\config\agent_config.yaml"
+          }
+          if ( "${{ matrix.MODE }}" -eq "gateway" ) {
+            Test-Path -Path "\tmp\splunk-support-bundle\config\gateway_config.yaml"
+          }
+
+  windows-choco-validation:
```

Comment:
Instead of building the choco package twice, let's move the choco build steps to a separate `windows-choco` that is dependent on `windows-msi` and a requirement for `windows-choco-validation`.

Reply:
Added `windows-choco` job.